Neil summmer21 dev

neilhatfield opened the pull request:
This pull request will bring the app into alignment with the current Style Guide.

As you review, please pay extra attention to:

+ The wording used on each page
+ Any glitches in the plots
+ The time animation (is the slider too fast/too slow for the plot updates?)
+ All standard/regular review criteria

neilhatfield committed:
Update 1-Summer 21

This update begins to tackle bring the app into alignment with the Style Guide (see #1 ).

This particular commit
Fixes #3
Fixes #4

neilhatfield committed:
Update 2-Summer '21

Restructured two data sets in relation to #5
Reworked code for Survival and Fecundity Pages

neilhatfield committed:
Update 3- Summer '21

Altered how the country-sex data is read in for the cohort pyramid. I also created a working prototype of the potential replacement plot

neilhatfield committed:
Mass Updates

Push of numerous updates

neilhatfield committed:
Update app.R

Removed extra packages and updated the references page.

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,843 @@
+          tabName = "overview",
+          h1("Life Tables"),
+          p("In this app, you will explore various ways to visualize life tables
+            for 3 different countries--China, the United Kingdom, and the United
```
Is there any way to make this an em-dash instead of --? I think it would look a little better

neilhatfield replied:
I put a tweak in place to see if the em-dash would work. Let me know what you think.

shravanisamala replied:
Looks great!

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,843 @@
+            metrics associated with life tables (e.g., survival and fecundity
+            rates) by country and by sex."),
+          h2("Instructions"),
+          p("Use the left sidebar menu to explore the following life tables."),
```
Should this end with a colon instead of a period since it says "the following"?

neilhatfield replied:
I've opted to remove "following".

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,843 @@
+          p("Use the left sidebar menu to explore the following life tables."),
+          tags$ul(
+            tags$li(tags$strong("Survival Rate:"), " Click the country and sex
+                    combination you preferred to compare, and view the distinct
```
"you prefer", everything else seems to be in present tense

neilhatfield replied:
Fixed.

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,843 @@
+            tags$li(tags$strong("Population Pyramids Through Time:"), " Click the start
+                    button to watch the population change by year in United
+                    States and United Kingdom."),
+            tags$li(tags$strong("Fecundity Rate:"), " Click the country you
```
The word "Fecundity" was also used in the info button, but I have no idea what it means with out googling it. Maybe there could be a place (either in the info button or the overview page itself) where the definition could be provided so that kids not familiar with or not in the bio-stats option could more easily understand!

shravanisamala replied:
Nevermind, I see that the definition is provided in the Fecundity Rate tab!

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,843 @@
+                column(
+                  width = 8,
+                  offset = 0,
+                  plotOutput("upperPyramid"),
```
I think a br() should be inserted here because at first glance, Percentage alive looks like the title for the graph below. Also, I don't know if this is doable, but is there a way to have the graphs be displayed side by side? I usually find that it is easier to compare graphs that are side by side. In this case it is a bit harder especially because the student has to keep scrolling up and down to compare.

neilhatfield replied:
I'm putting a space to see how that functions. However, I'm noticing that the legend is currently putting in all six pairings. When I was working yesterday, only the two pairings used in each graph were showing. I'm not sure what triggered the change.

shravanisamala replied:
Looks better! And as for the legends, when I pull the code now it still only shows the two that are in each pyramid plot.

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,843 @@
+              title = "Country & Sex Comparisons",
+              br(),
+              p("Population pryamids can also be mixed and matched across countries
+                and sex. Create various pryamids to compare the different pairings
```
pyramids is misspelled here

neilhatfield replied:
Fixed.

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,843 @@
+                  width = 4,
+                  offset = 0,
+                  wellPanel(
+                    p("Select which Country-Sex pairings you want to compare."),
```
Does the style guide allow you to bold likes 219, 222, and 235? These labels kind of blend into the rest of the text and I think it would be nice to have them stand out in some way since they are not part of the list of things to compare.

neilhatfield replied:
Generally, you want to try to limit usages of bold (`strong`) and italics (`em`) so that they retain their special case status. Over using them can dilute their meaning. If we were to be extremely judicious with our usage we might be able to do something, but we would need something specific. Making a whole label bold seems like a step too far. Perhaps there's something else we can do to try to resolve the issue you're describing.

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,843 @@
+          p("Connecting population pyramids with time allows us to explore changes
+            in a country's population. This allows us to see when there might be
+            large increases in the population as well as when a country might
+            suffer from catesphroic loss of life."),
```
catastrophic is misspelled

neilhatfield replied:
Fixed.

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,843 @@
+            column(
+              width = 8,
+              offset = 0,
+              plotOutput("timePyramid")
```
I see at the bottom it says "Blue is male, Red is female." The other graphs have a legend either inside the graph or outside the graph, so I think to stay consistent adding a legend would be nice!

neilhatfield replied:
Part of the issue is that the other pyramids are drawing upon a fixed country-sex definition for the coloration. This pyramid doesn't draw from that pooling, sticking with the same coloring regardless of which country. We can certainly explore a legend option.

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,843 @@
+            column(
+              width = 8,
+              offset = 0,
+              plotOutput("timePyramid")
```
I also really like the added feature of getting to press the pay button and the graph automatically increases by year. Is there any way to make it go faster or allow the student to choose a speed?

neilhatfield replied:
~No, there isn't a way for the user to set the speed.~ Well, there might be a hack job way to make some user control but I'm not sure how viable that would be. 

Currently the animation is set to pause 2000 milliseconds between ticks. Is this too long? We run into an issue where if the wait time is too short the slider will get ahead of the plot by multiple ticks.

shravanisamala replied:
I see! The speed isn't a big issue at all the animation still looks great!

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,843 @@
+            column(
+              width = 8,
+              offset = 0,
+              plotOutput("timePyramid")
```
I noticed that the other pyramid's y-axis goes up to a hundred but this one only goes up to 85. Even if there aren't any ppl that live up to that age, I think going to 100 as the y-axis limit would be better so it is similar to the other pages

neilhatfield replied:
Fixed. I also added a note about interpreting the oldest age bar as being that age and older.

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,843 @@
+        tabItem(
+          tabName = "fecundityRate",
+          h2("Fecundity Rate"),
+          p("A common metric for life tables is that a country's fecundity rate.
```
"A common metric for life tables is a country's fecundity rate."

neilhatfield replied:
Fixed

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,843 @@
+              )
+            ) #original link is broken: http://bxjg.circ.gov.cn/web/site0/tab5216/info4054990.htm
+          ),
+          br(),
```
The page looked like it had awkward white space here so I would remove one of two br()s.

neilhatfield replied:
Currently, we put three `br()` between the last element and the `copyrightInfo` call.

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,843 @@
+            br(),
+            br(),
+            br(),
+            div(class = "updated", "Last Update: 7/7/2021 by NJH.")
```
For some reason, when I scroll down after this there is a large white space.

neilhatfield replied:
I don't know what you mean by this.

shravanisamala replied:
It's no longer there, so no worries! Yesterday the page kept scrolling way past the updated statement so there was just a bunch of empty space before I could stop scrolling even though there was nothing after this statement on the overview page.

shravanisamala reviewed (changes requested):
Hi Dr. Hatfield! The app looks great! I caught a few typos so I mentioned those in the comments below. I had no problems with the graphs (I think I tried most if not all of the combinations). The wait time for them to pop up was very short and I really liked the idea to include that animation. If I had one overall comment, it would be to consider including buttons that move students forward though the app instead of having to click on the side tabs. Especially if the student minimizes the left side bar, they may be a little lazy (like me whoops) or annoyed that they have to keep opening up the side bar to navigate instead of having buttons that just keep going to the next page.

neilhatfield committed:
Update app.R

Updates incorporating Shravni's suggestions.

neilhatfield commented:
> If I had one overall comment, it would be to consider including buttons that move students forward though the app instead of having to click on the side tabs. Especially if the student minimizes the left side bar, they may be a little lazy (like me whoops) or annoyed that they have to keep opening up the side bar to navigate instead of having buttons that just keep going to the next page.

I'm not particularly enamored with putting navigational buttons on every page just because. The whole point of the side bar menu is to serve as the navigation hub. (I'm becoming less and less fond of the starting button on the overview page.)

If a user needs to work through the app in a particular order (e.g., see the Diagram Wizard of the Hasse Diagram app), then navigational buttons make more sense–especially when the navigation is purposefully hidden.

paridhi1603 reviewed (commented):
Hi,
I just went through the app and I feel it looks great. Although I found a few things like when I was going through the Fecundity Rate and Survival Rate, the name of the countries appear twice showing a different color scheme and going out of the application window(not properly getting displayed). Moreover, when I check some country's name off, the check box disappears, maybe we can put a dot to show it is selected rather than make it disappear? I have attached some screenshots to show the issue. 

<img width="796" alt="Screen Shot 2021-07-11 at 6 11 22 PM" src="https://user-images.githubusercontent.com/56276877/125211307-73644680-e273-11eb-9df3-bfdd8d9fd172.png">
<img width="1177" alt="Screen Shot 2021-07-11 at 6 11 35 PM" src="https://user-images.githubusercontent.com/56276877/125211309-73fcdd00-e273-11eb-85f1-a99bb468dabb.png">

neilhatfield committed:
Update app.R

Fixed the double legend on the survival rate plot

neilhatfield commented:
>  Moreover, when I check some country's name off, the check box disappears, maybe we can put a dot to show it is selected rather than make it disappear?

I fixed the double legend issue and am now trying to track down the disappearing checkbox. What browser were you using?

paridhi1603 commented:
I am using Google Chrome but I viewed in on R itself

shravanisamala commented:
@neilhatfield Hi Dr. Hatfield, I was just looking through the code and I saw that on the Survival rate page, the header Survival rate is misspelled!

neilhatfield commented:
> I was just looking through the code and I saw that on the Survival rate page, the header Survival rate is misspelled!
@shravanisamala  Good catch. I've fixed that and will include it in the next push.

neilhatfield committed:
Spelling Fix

JingFu1 committed:
Update app

solve issue#6 and update the plots of Time & Pop. Pyramids page

JingFu1 committed:
README.md

add screenshot

JingFu1 committed:
update

icon update

JingFu1 committed:
latest

solved key problem

JingFu1 committed:
ToDoList

tarynmch reviewed (changes requested):
I think this app is just almost ready to deploy, I just have a few edits and suggestions. The first thing I noticed was that there is no cite as on the bottom of the overview. Also, I know there needs to be an added author to the description file and just small things like dates should be updated.  I also have a few suggestions. I know this app is being used in the classroom, so I suggest some sort of prereq, maybe what the shape of a population pyramid means (growth, decay, or constant). In the Time and Pop. Pyramids I suggest that there should be a formal legend instead of words to keep it concise with the other graphs. Also, if possible, make the play button bigger or explain in the description its purpose because I also missed that it was there completely. I don't see any functionality issues and think the app just needs some minor adjustments to be ready to deploy.

Sburke26 reviewed (changes requested):
Functionality-wise, I did not notice any issues. Here are some small things I did notice while reviewing:

The year badge in the README is not properly updated.
The description of the app in the README is missing the Heading of 'App Description'.
There is not a 'Cite this app as...' entry in the overview.
The front page button should use a lightning bolt to send to any type of activity page.
There are some spacing/indentation issues in the code. 
Warning in the console: "Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0. Please use `linewidth` instead. This warning is displayed once every 8 hours. Call `lifecycle::last_lifecycle_warnings()` to see where this warning was generated."

Small suggestions:

I feel a prerequisites page would be a beneficial addition. (instead of putting this information on each page).
I like how two of the activity pages have a question for the user to think about while traversing the page. Maybe add questions for the other two activity pages.
The Time and Population Pyramids does not have a proper legend.
The colors of the Fecundity Rate do not really contrast well. the 
There are two plots with alt arguments that are more general than the other plots. (The other plots' alt text updates with the selected inputs). It may be appropriate to update these two to be more specific like the rest.

tarynmch committed:
app.R and details update

Updated app.R (console issue), description, and README file

tarynmch committed:
Grammar and Spacing

tarynmch committed:
Update app.R

tarynmch committed:
Update app.R

tarynmch committed:
Update app.R

references

Sburke26 reviewed (changes requested):
Regarding my previous comment about the legend of the Time and Population Plots, I now that the data in the allData.RData file does not support the ability to create a legend like the other plots. Therefore, I think it is fine to be left as is.

In terms of issues, the year badge in the README should have the year in the square brackets where it says "year".  Also, the information button is currently under the title of "Instructions". It may be less redundant to change the information button title to "Information" instead of having two instruction sections. 

Other than that, I could not find other issues, and it appears that the app is just about ready.
Good work!

tarynmch committed:
post review

tarynmch committed:
final updates

tarynmch committed:
Update README.md

tarynmch committed:
Update app.R

acknowledgments

Sburke26 reviewed (dismissed):
I didn't notice any issues since the updates after my last review. The app functions well. Good work!

neilhatfield committed:
Final Coding Modifications

Small coding changes and updates; upgrading the app from Experimental to Maturing

neilhatfield commented:
@tarynmch Final Checks:

1. Pull my updates and redeploy to the testing site.
2. Email Dennis for him to double check
3. Check the [App's Issues](https://github.com/EducationShinyAppTeam/Life_Table/issues) and see which may be closed with this pull request. Add all resolved issues to the Development menu (right-hand column of the Pull Request).

tarynmch committed:
Updating Data

Add links and to do so data can be updated

tarynmch committed:
post reviews

suggestions from prof. pearl

neilhatfield committed:
Final Updates
